var/spack/repos/builtin/packages/rust-bootstrap/package.py

```diff
@@ -133,6 +133,12 @@ class RustBootstrap(Package):
         if os in rust_releases[release] and target in rust_releases[release][os]:
             version(release, sha256=rust_releases[release][os][target])
 
+    # rust-ldd and libLLVM both depend on zlib, which is not vendored.
+    depends_on("zlib-api")
+    depends_on("zlib-ng +shared", when="^[virtuals=zlib-api] zlib-ng")
+    depends_on("zlib +shared", when="^[virtuals=zlib-api] zlib")
```

Comment:
Not a blocker on this, but we should really fix variants on virtuals so that you could say `zlib-api+shared`. (Although yes it's possible `+shared` doesn't exist on the provider package -- so maybe something like `zlib-api++shared` for variant propagation should work.)